nomenclature/codelist.py

```diff
@@ -542,3 +543,28 @@ def from_directory(cls, name: str, path: Path, file_glob_pattern: str = "**/*"):
             mapping[code.name] = code
 
         return cls(name=name, mapping=mapping)
+
+    def filter(self, hierarchy: str) -> "RegionCodeList":
+        """A filter that returns the components within a provided hierarchy.
+
+        Parameters
+        ----------
+        hierarchy : str
+            The name of the hierarchy whose components you are looking for.
+
+        Raises
+        ------
+        ValueError
+            If User inputs a typo or a hierarchy not compatible with the given model.
+
+        Returns
+        -------
+        RegionCodeList
+            Returns a list of the component regions of the inputted hierarchy.
+        """
+
+        mapping = {k: v for (k, v) in self.mapping.items() if v.hierarchy == hierarchy}
+        if mapping:
+            return RegionCodeList(name=f"{hierarchy}", mapping=mapping)
```

Comment:
`hierachy` is already a string and you're not concatenating it with anything else so no need for an f-string here:
```suggestion
            return RegionCodeList(name=hierarchy, mapping=mapping)
```

On a related note I'm not sure if we should "rename" the resulting RegionCodeList. This `RegionCodeList(name=self.name, mapping=mapping)` might be fine as well but that's more of a design question.